numba/parfor.py

```diff
@@ -3162,14 +3178,568 @@ def __init__(self, blocks):
 
 postproc.ir_extension_insert_dels[Parfor] = parfor_insert_dels
 
+def find_args_written_to(block, arg_aliases):
+    """
+    Scan all the blocks to find arrays written to.
+    """
+    for stmt in block.body:
+        if isinstance(stmt, ir.StaticSetItem) or isinstance(stmt, ir.SetItem):
+            if stmt.target.name in arg_aliases:
+                if config.DEBUG_ARRAY_OPT >= 1:
+                    print("Found argument alias written to for array", stmt.target.name)
+                return True
+    return False
+
+def duplicate_stmt(x, typemap, calltypes, dup_typemap, dup_calltypes, label_offset, toprint=""):
```

Comment:
Can you add docstring to describe this function?

Btw, if it is a general purpose statement duplicator, it can be moved to the `ir_utils.py`.